middleware/comm/server.py

```diff
@@ -81,15 +82,20 @@ def handle(self, client_address, rfile, wfile):
             raise RequestHandlerError(message)
         finally:
             output = json.dumps(response, sort_keys=True)
-            self.logger.info("=> [%s]: %s", client_address, output)
-            self._reply(wfile, output)
+            success = self._reply(wfile, output)
+            if success:
+                self.logger.info("=> [%s]: %s", client_address, output)
 
     def _reply(self, wfile, output):
+        success = False
         try:
             wfile.write(output.encode(self.ENCODING))
             wfile.write("\n".encode(self.ENCODING))
+            success = True
```

Comment:
This is reporting a code scanning issue. Can we just `return true` at the end of the `try` block and `return false` after the logging on the `except` block? Wouldn't that achieve the same without the need for a variable?

Reply:
Thanks for the suggestion, this is fixed now